x/concentrated-liquidity/lp_test.go

```diff
@@ -551,6 +551,95 @@ func (s *KeeperTestSuite) TestIsInitialPosition() {
 	}
 }
 
+func (s *KeeperTestSuite) TestUpdatePosition() {
+	type updatePositionTest struct {
+		poolId          uint64
+		owner           sdk.AccAddress
+		lowerTick       int64
+		upperTick       int64
+		liquidityDelta  sdk.Dec
+		amount0Expected sdk.Int
+		amount1Expected sdk.Int
+		expectedError   bool
+	}
+
+	tests := map[string]updatePositionTest{
+		"happy path": {
+			poolId:          1,
+			owner:           s.TestAccs[0],
+			lowerTick:       DefaultLowerTick,
+			upperTick:       DefaultUpperTick,
+			liquidityDelta:  DefaultLiquidityAmt,
+			amount0Expected: DefaultAmt0Expected,
+			amount1Expected: DefaultAmt1Expected,
+			expectedError:   false,
+		},
+		"error: invalid pool id": {
+			poolId:          2,
+			owner:           s.TestAccs[0],
+			lowerTick:       DefaultLowerTick,
+			upperTick:       DefaultUpperTick,
+			liquidityDelta:  DefaultLiquidityAmt,
+			amount0Expected: DefaultAmt0Expected,
+			amount1Expected: DefaultAmt1Expected,
+			expectedError:   true,
+		},
+		"new position when calling update position": {
+			poolId:          1,
+			owner:           s.TestAccs[1],
+			lowerTick:       DefaultLowerTick,
+			upperTick:       DefaultUpperTick,
+			liquidityDelta:  DefaultLiquidityAmt,
+			amount0Expected: DefaultAmt0Expected,
+			amount1Expected: DefaultAmt1Expected,
+			expectedError:   false,
+		},
+	}
+	for name, tc := range tests {
+		s.Run(name, func() {
+			tc := tc
+			s.SetupTest()
+
+			// create a CL pool
+			originalPool := s.PrepareConcentratedPool()
+
+			// create position
+			// Fund test account and create the desired position
+			s.FundAcc(s.TestAccs[0], sdk.NewCoins(sdk.NewCoin(ETH, DefaultAmt0), sdk.NewCoin(USDC, DefaultAmt1)))
+			_, _, _, err := s.App.ConcentratedLiquidityKeeper.CreatePosition(s.Ctx, originalPool.GetId(), s.TestAccs[0], DefaultAmt0, DefaultAmt1, sdk.ZeroInt(), sdk.ZeroInt(), tc.lowerTick, tc.upperTick)
+			s.Require().NoError(err)
+
+			// system under test
+			actualAmount0, actualAmount1, err := s.App.ConcentratedLiquidityKeeper.UpdatePosition(
+				s.Ctx,
+				tc.poolId,
+				tc.owner,
+				tc.lowerTick,
+				tc.upperTick,
+				tc.liquidityDelta,
+			)
+
+			if tc.expectedError {
+				s.Require().Error(err)
+			} else {
+				s.Require().NoError(err)
+				s.Require().Equal(actualAmount0, tc.amount0Expected)
+				s.Require().Equal(actualAmount1, tc.amount1Expected)
+
+				// validate if position has been properly updated
+				s.validatePositionUpdate(s.Ctx, tc.poolId, tc.owner, tc.lowerTick, tc.upperTick, tc.liquidityDelta)
+
+				// validate if pool liquidity has been updated properly
+				poolI, err := s.App.ConcentratedLiquidityKeeper.GetPoolById(s.Ctx, tc.poolId)
+				s.Require().NoError(err)
+				concentratedPool := poolI.(types.ConcentratedPoolExtension)
+				s.Require().Equal(originalPool.GetLiquidity().Add(tc.liquidityDelta).Add(tc.liquidityDelta), concentratedPool.GetLiquidity())
```

Comment:
Why do we add liquidityDelta twice to the pools liquidity?

Reply:
this is happening because first time by createPosition, second time by updatePosition: I will leave additional in line comments to make this clearer do you think this makes sense?

Reply:
Yeah it LGTM now, thank you